src/software/thunderscope/replay/proto_player.py

```diff
@@ -130,6 +134,91 @@ def is_log_entry_corrupt(log_entry: bytes, version: int) -> bool:
+         - Start timestamp: timestamp of the first log entry in the chunk
+         - Filename: replay chunk file name (%d.reply)
+
+        :param folder_path: folder containing all the replay files
+
+        :return: a dictionary for mapping replay filename to the start
+            timestamp of the first entry in the chunk.
+
+        """
+        chunk_indices: dict[str:float] = dict()
+        for chunk_name in self.sorted_chunks:
+            chunk_data = ProtoPlayer.load_replay_chunk(chunk_name, self.version)
+            if chunk_data:
+                start_timestamp, _, _ = ProtoPlayer.unpack_log_entry(
+                    chunk_data[0], self.version
+                )
+                chunk_indices[os.path.basename(chunk_name)] = start_timestamp
+        if chunk_indices:
+            with open(
+                os.path.join(folder_path, ProtoPlayer.CHUNK_INDEX_FILENAME), "w"
+            ) as index_file:
+                index_file.write(f"Generated on {time.time()}\n")
+                for key, value in chunk_indices.items():
+                    index_file.write(f"{value}, {key}\n")
+            logging.info("Created chunk index file successfully.")
+        else:
+            logging.warning(f"Failed to build chunk index for {folder_path}")
+        return chunk_indices
+
+    def is_chunk_indexed(self) -> bool:
+        """Returns true if the chunk index is already built.
+
+        :return: if the chunk index file exists
+        """
+        return os.path.exists(
+            os.path.join(self.log_folder_path, ProtoPlayer.CHUNK_INDEX_FILENAME)
+        )
+
+    def load_chunk_index(self) -> dict[str:float]:
+        """Loads the chunk index file.
+
+        :return: the chunk indices.
+        """
+        chunk_indices: dict[str:float] = dict()
+        try:
+            with open(
+                os.path.join(self.log_folder_path, ProtoPlayer.CHUNK_INDEX_FILENAME),
+                "r",
+            ) as index_file:
+                # skip the first timestamp line
+                index_file.readline()
+
+                for line in index_file:
+                    start_timestamp, chunk_name = line.split(",")
+                    start_timestamp = float(start_timestamp)
+                    chunk_name = chunk_name.strip()
+                    chunk_indices[chunk_name] = start_timestamp
+            logging.info("Pre-existing chunk index file found and loaded.")
+        except Exception as e:
+            logging.warning(f"An Exception occurred when loading chunk index file {e}")
+        return chunk_indices
+
+    def get_chunk_index(self) -> dict[str:float]:
+        """Returns the chunk indices.
+        NOTE: if the chunk index was not built, this function will automatically build one.
+
+        :return: chunk indices.
+        """
+        if not self.is_chunk_indexed():
+            return self.build_chunk_index(self.log_folder_path)
+        else:
```

Comment:
I think you could take out this else statement. Since you have return on top. This would reduce the amount of indenting to make it nicer to look at.

Reply:
Done